### ISSUE TITLE
[refactoring] idea: convert a ternary operator into an if statement and vice-versa

### ISSUE THREAD

**Issue by hyp** [llvm/llvm-project#34348]
|  |  |
| --- | --- |
| Bugzilla Link | [35000](https://llvm.org/bz35000) |
| Version | trunk |
| OS | All |
| CC | @TH3CHARLie |

## Extended Description 
This is a possible idea for a new refactoring action for clang-refactor & editors that support integration with Clang's refactoring engine.

This idea suggests adding a refactoring that can convert a ternary operator into an if statement and vice-versa. For example, this statement:
```cpp
  int x = expression ? a : b;
```
Can become:
```cpp
    int x;
    if (expression) {
        x = a;
    } else {
        x = b;
    }
```
and vice-versa.

**Comment by elhewaty** [comment]
Hello, I am a beginner who is interested in contributing to LLVM.
Can anyone guide me through this?

**Comment by rachejazz** [comment]
So I see many opted for working on this before, but as I read on other issues, as long as I send a patch I can get a review?

**Comment by EugeneZelenko** [comment]
@rachejazz: Sure, patch is best way to proceed!

**Comment by elhewaty** [comment]
@rachejazz, Yes, and feel free to work on it, my comment was a long time ago.
The sign of work is the assignee tag and the pull requests.

**Comment by bhanuponguru** [comment]
Hi,

I would like to work on this issue, I request you to assign this issue to me.

Thank you.
